yt_dlp/extractor/tiktok.py

```diff
@@ -431,13 +441,32 @@ def _parse_aweme_video_web(self, aweme_detail, webpage_url, video_id):
 
         for download_url in traverse_obj(video_info, (('downloadAddr', ('download', 'url')), {url_or_none})):
             formats.append({
+                **watermark_info(download_url),
                 'format_id': 'download',
                 'url': self._proto_relative_url(download_url),
                 'ext': 'mp4',
                 'width': width,
                 'height': height,
             })
 
+        for bitrate_info in traverse_obj(video_info, ('bitrateInfo', lambda _, v: v['PlayAddr']['UrlList'])):
+            format_info = traverse_obj(bitrate_info, {
+                'format_id': (('GearName', {str}), ('QualityType', {int}, {str_or_none}), any),
+                'vcodec': ('CodecType', {str}),
+                'tbr': ('Bitrate', {functools.partial(int_or_none, scale=1000)}),
+                'filesize': ('PlayAddr', 'DataSize', {int_or_none}),
+                'width': ('PlayAddr', 'UrlKey', {lambda x: re.search(r'_(\d+)p_', x)}, 1, {int_or_none}),
+            })
+            format_info['height'] = try_call(lambda: int(format_info['width'] / 0.5625))
```

Comment:
can we calculate the ratio by `width` / `height`?
tiktok actually have videos in different aspect ratio:
https://www.tiktok.com/@awilern/video/7357655658960801031
https://www.tiktok.com/@seventeen17_official/video/7365107718765513985
https://www.tiktok.com/@bongbong_lover/video/7362476954949373190